docker-images/kafka/scripts/kafka_config_generator.sh

```diff
@@ -150,9 +150,15 @@ if [ "$KAFKA_EXTERNAL_ENABLED" ]; then
       LISTENER_NAME_EXTERNAL_SSL_CLIENT_AUTH="none"
     fi
 
+    if [ -n "$KAFKA_CUSTOM_TLS_CERT" ] && [ -n "$KAFKA_CUSTOM_TLS_KEY" ]; then
+      LISTENER_NAME_EXTERNAL_SSL_KEYSTORE_LOCATION="/tmp/kafka/custom.keystore.p12"
+    else
+      LISTENER_NAME_EXTERNAL_SSL_KEYSTORE_LOCATION="/tmp/kafka/cluster.keystore.p12"
+    fi
+
     EXTERNAL_LISTENER=$(cat <<EOF
 # EXTERNAL interface configuration
-listener.name.external.ssl.keystore.location=/tmp/kafka/cluster.keystore.p12
+listener.name.external.ssl.keystore.location=${LISTENER_NAME_EXTERNAL_SSL_KEYSTORE_LOCATION}
```

Comment:
if we are exposing this feature for the `tls` listener type as well (which is for applications connecting from inside the same Kubernetes cluster via TLS) then the generated keystore should be used in the "TLS listener" section as well.